### PR TITLE
ip address added in audit entries

### LINKED ISSUE #714: Include IP address in audit log?
Is it possible to include the client IP address in the audit log?

This would enhance security, if we were able to do it.

Will require change(s) to audit log in backend.  But, field must be optional, since I can't provide it for legacy audit log entries.

### PR DESCRIPTION
**Pull request by TahaKhanAbdalli**
fixes #714

### CODE REVIEW

File: src/utils/helper.ts [hunk 1]
```diff
@@ -17,3 +18,24 @@ export const checkIfUserIsActive = (user: User): boolean => {
   }
   return true
 }
+
+const getIp = async (): Promise<string | undefined> => {
+  try {
+    const data = await axios.get('https://api.ipify.org?format=json')
```
Comment (IanMayo) [inline src/utils/helper.ts:24]: Sorry @TahaKhanAbdalli - the clients will be running this on a network without an Internet connection, so we can't make the above call.

Comment (TahaKhanAbdalli) [reply src/utils/helper.ts:24]: Then, there's no easy way to fetch the ip with vanilla js

Comment (IanMayo) [reply src/utils/helper.ts:24]: Ok. Could you investigate how to do it with node please?  `Soul` supports _extensions_, which are blocks of server-side NodeJS code.  Maybe we can develop a Soul extension to give us the caller's IP address.

File: src/utils/helper.ts [hunk 2]
```diff
@@ -17,3 +18,24 @@ export const checkIfUserIsActive = (user: User): boolean => {
   }
   return true
 }
+
+const getIp = async (): Promise<string | undefined> => {
+  try {
+    const data = await axios.get('http://localhost:8000/api/ip')
```
Comment (IanMayo) [inline src/utils/helper.ts:24]: I'm pretty sure we need to take the port from the env variable

**Comment by IanMayo** [comment]
Thanks @TahaKhanAbdalli - we also need to handle the populating the $EXTENSION variable.  _Maybe_ in the `.env`?

File: src/utils/helper.ts [hunk 3]
```diff
@@ -17,3 +18,26 @@ export const checkIfUserIsActive = (user: User): boolean => {
...
 }
+
+const getIp = async (): Promise<string | undefined> => {
+  try {
+    const data = await axios.get(
+      `${process.env.VITE_API_URL ?? 'http://localhost:8000/api'}/ip`
+    )
+    return data.data?.ip
+  } catch (error) {
+    return undefined
```
Comment (IanMayo) [inline src/utils/helper.ts:29]: should we log any errors to the console @TahaKhanAbdalli ?

**Review by IanMayo — APPROVED** [review]

**Pull request merged**